lib/compilers/argument-parsers.js

```diff
@@ -142,17 +144,55 @@ export class ClangParser extends BaseParser {
             compiler.compiler.supportsIrView = true;
             compiler.compiler.irArg = ['-Xclang', '-emit-llvm', '-fsyntax-only'];
         }
+        //console.log("---->", this.mllvmOptions, "<----");
+        //console.log("------------------>", BaseParser.hasSupport(options, '-mllvm'),
+        //    this.mllvmOptions.has('-print-before-all'), this.mllvmOptions.has('-print-after-all'));
+        if (
+            BaseParser.hasSupport(options, '-mllvm') &&
+            this.mllvmOptions.has('--print-before-all') &&
+            this.mllvmOptions.has('--print-after-all')
+        ) {
+            compiler.compiler.supportsLLVMOptPipelineView = true;
+            compiler.compiler.llvmOptArg = ['-mllvm', '--print-before-all', '-mllvm', '--print-after-all'];
+            compiler.compiler.llvmOptModuleScopeArg = [];
+            if (this.mllvmOptions.has('--print-module-scope')) {
+                compiler.compiler.llvmOptModuleScopeArg = ['-mllvm', '-print-module-scope'];
+            }
+            //console.log("--------->", compiler.compiler.llvmOptArg, compiler.compiler.llvmOptModuleScopeArg);
+        }
 
         if (BaseParser.hasSupport(options, '-fcolor-diagnostics')) compiler.compiler.options += ' -fcolor-diagnostics';
         if (BaseParser.hasSupport(options, '-fno-crash-diagnostics'))
             compiler.compiler.options += ' -fno-crash-diagnostics';
     }
 
+    static tempFileId = 0;
+
     static async parse(compiler) {
+        //console.log("=================================> DO PARSE");
         const options = await ClangParser.getOptions(compiler, '--help');
+        const filename = `blank${ClangParser.tempFileId++}.cpp`;
+        const handle = await fs.open(filename, 'w');
```

Comment:
no path means this will certainly fail on prod, you can put it in /tmp somewhere